_docs/for-contributors/2017-01-01-building-from-source.md

```diff
@@ -205,3 +205,113 @@ Now, to run the wallet connected to the Cardano SL in dev-mode, call:
 ```
 CARDANO_API=true npm run dev
 ```
+
+## Building Cardano SL node for Testing Purpose
+
+If you want to test functionality of Cardano SL node in our [testnet](/timeline/testnet/), you should go to the root directory of the repository and switch to release branch, for example, `cardano-sl-0.2`. After that build a node with this command:
+
+~~~ bash
+CSL_SYSTEM_TAG=linux64 stack build --flag cardano-sl:-dev-mode
+~~~
+
+It means that you build a node in **production** mode.
+
+Now run your node with this command:
+
+~~~ bash
+stack exec -- cardano-node \
+    --db-path node-db0 \
+    --rebuild-db \
+    --dht-key a_P8zb6fNP7I2H54FtGuhqxaMDAwMDAwMDAwMDAwMDA= \
+    --listen 127.0.0.1:3000 \
+    --json-log logs/2017-03-10_133948/node0.json \
+    --logs-prefix logs/2017-03-10_133948 \
+    --log-config logs.yaml \
+    --kademlia-dump-path logs/2017-03-10_133948/dump/kademlia0.dump \
+    --wallet \
+    --keyfile secret.key \
+    --peer 35.156.234.240:3000/S-GJ1HDZyHpIS_ZOFBlZsipkGNXy-tmCDtY_CvaiywE= \
+    --peer 35.157.148.244:3000/nzcEO6Cv7tKD3REWoWc9vcHMOZEQWFGLX4150bt_cYw= \
+    --peer 52.52.249.66:3000/dYGuDj0BrJxCsTC9ntJE7ePT7wUoVdQMH3sKLzQD8bo= \
+    --peer 52.8.23.163:3000/8hVd3J5Jc0GCkIV-QeiOkwDqH38QooYGo3iHJnZuIk4=
+~~~
+
+You can download `secret.key` from [Cardano Test-Net Faucet](https://tada.iohk.io).
```

Comment:
Some clarification as to where this key goes to would be good, i.e., to the project's root folder, or somewhere else.